### ISSUE TITLE
Inconsistent quotation marks in elasticsearch.yml

### ISSUE THREAD

**Issue by ak0ska** [elastic/puppet-elasticsearch#705]
* Module version: 0.13.2
* Puppet version: 3.8.7
* OS and version: CentOS 7.2

## Bug description
Hello,

This is probably a nitpicky issue, but here it goes: with the current version of the module the generated elasticsearch.yml uses quotation marks around numbers inconsistently. I think it's due to the `to_yaml` call in the template, and how Puppet interprets the types of its own variables. We previously used version 0.10.1 of the module, where this was not an issue.

Here's an example:

``` puppet
class{ 'elasticsearch':
    [..]
    config           => {
      'cluster' => {
        'routing.allocation' => {
          'node_concurrent_recoveries'        => 20,
          'node_initial_primaries_recoveries' => floor(30),
          'awareness.attributes'              => 'landb_parent'
        }
    }
}
``` 

This will result in the following entries in the yaml file:
``` yaml
cluster.routing.allocation.awareness.attributes: landb_parent
cluster.routing.allocation.node_concurrent_recoveries: "20"
cluster.routing.allocation.node_initial_primaries_recoveries: 30
```

As you can see the string `landb_parent` has not quotes around it, while the value 20 has. Number 30 has been put through Puppet's `floor` function, and it has no quotes around.

In our config we have many more numerical values, some are calculated using arithmetic operators, or put through numerical functions like `floor`. These values have no quotes around them. However the "static" numerical values (like the 20 in the example) get quoted in the config file.

It does not look pretty. :) Apart from that, does ElasticSearch makes a difference between quoted and unquoted values when parsing the settings file?

Thanks,
Akos

**Comment by tylerjl** [comment]
As far as I understand, Elasticsearch should parse either form of those numerics the same way. You're correct in assuming it's the way puppet handles hashes internally that is causing the inconsistency here, I've had to work around it a few times to ensure numbers are represented correctly in a few places (it may have something to do with pson, not sure).

For example, when passing a Puppet hash to create a new index template, [I have to explicitly call a custom function called `deep_to_i`](https://github.com/elastic/puppet-elasticsearch/blob/a93350b4b13ad5c8f795c67a3ca3046a88cdf25a/lib/puppet/type/elasticsearch_template.rb#L45) in order to ensure all numbers are actually numbers before comparing what puppet wants the template to be to what the Elasticsearch API reports. [`deep_to_i` is here](https://github.com/elastic/puppet-elasticsearch/blob/a93350b4b13ad5c8f795c67a3ca3046a88cdf25a/lib/puppet_x/elastic/deep_to_i.rb). It's definitely not ideal, but it solves the problem.

In this case I agree that it's probably more cosmetic than a bug, but it good to note it here at least. There's more pressing items on the agenda (like compatability with 5.x) but I'd wholeheartedly accept PRs if someone wanted to tackle it.

In addition to some tests that reproduce the quoted numerics, the solution is pretty straightforward and probably just needs to be a call to `deep_to_i` in the template (nearby the `to_yaml` call).

**Comment by tylerjl** [comment]
Fixed in #715
